rmk-macro/src/layout.rs

```diff
@@ -37,6 +37,43 @@ fn expand_row(row: Vec<String>) -> TokenStream2 {
     quote! { [#(#keys), *] }
 }
 
+/// Get modifier combination, in types of mod1 | mod2 | ...
```

Comment:
Please add documentation here to specify what the five `bool`s represent. And there's a minor format issue: 
`){` -> `) {`

Reply:
I decided to use a struct and refactor away the tuples. Also i implemented the macro expansion in the struct impl. Let me know if I did something wrong, I'm not expert on macros. The code compile, I'll test it now on the hardware.

Reply:
i confirm the hardware test is working for MT(A,LShift) .